### PR TITLE
Fix tablet intersection bug and start of Pointer migration work (WIP)

### PR DESCRIPTION
**Pull request by SamGondelman**
- cleans up some of the pointer/picking code
- fixes [FB7965](https://highfidelity.fogbugz.com/f/cases/7965/Laser-pressing-or-tablet-buttons-is-unreliable)

This is part of ongoing work to improve the UI performance, but since it was easier to fix the intersection bug in this branch we're merging this WIP to get the fix into master

Test plan:
- in HMD, hand lasers on the tablet should work much better for clicking
- mouse grab in desktop should still work
- hand lasers on HUD, web entities, and other entities should still behave the same as in master

### CODE REVIEW

**Comment by Nex-Pro** [comment]
awaiting build for WIN

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4212/HighFidelity-Beta-PR11612-2e69d2b1ad3576120aa917e462da5968fc59616d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4212/HighFidelity-Beta-PR11612-2e69d2b1ad3576120aa917e462da5968fc59616d.exe

**Comment by Nex-Pro** [comment]
TESTING

**Comment by Nex-Pro** [comment]
Awaiting fresh

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4216/HighFidelity-Beta-PR11612-ba4f41b59a2780826a75ea54880b398c9437fdce.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4216/HighFidelity-Beta-PR11612-ba4f41b59a2780826a75ea54880b398c9437fdce.exe

**Comment by Nex-Pro** [comment]
Still broken

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4218/HighFidelity-Beta-PR11612-50929b780d962eb35e76b2ad98b939036c9cff44.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4218/HighFidelity-Beta-PR11612-50929b780d962eb35e76b2ad98b939036c9cff44.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4223/HighFidelity-Beta-PR11612-975a3e5bbbbe9b93ca6758a0b1a65c9c4765ae36.dmg

**Comment by Nex-Pro** [comment]
T E S T I N G

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4223/HighFidelity-Beta-PR11612-975a3e5bbbbe9b93ca6758a0b1a65c9c4765ae36.exe

**Comment by Nex-Pro** [comment]
### PASS
Meets the tests.
Lasers are working well on the tablet. Scroll bars are good.
Editing entity properties with the lasers was fine.

Couple of odd laser issues not related to the tablet interactions.
Will see if can repro in master.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4242/HighFidelity-Beta-PR11612-58172cfbd493319a02d003cac5a0bd44f9dcddc0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4242/HighFidelity-Beta-PR11612-58172cfbd493319a02d003cac5a0bd44f9dcddc0.exe

File: interface/src/raypick/LaserPointer.cpp [hunk 1]
```diff
@@ -22,9 +22,10 @@ LaserPointer::LaserPointer(const QVariant& rayProps, const RenderStateMap& rende
     _faceAvatar(faceAvatar),
     _centerEndY(centerEndY),
     _lockEnd(lockEnd),
-    _distanceScaleEnd(distanceScaleEnd)
+    _distanceScaleEnd(distanceScaleEnd),
+    _rayPickUID(DependencyManager::get<RayPickScriptingInterface>()->createRayPick(rayProps))
```
Comment (ZappoMan) [inline interface/src/raypick/LaserPointer.cpp]: did you mean to not change this ? 
why is this not:  `qApp->getRayPickManager()`

Comment (SamGondelman) [reply interface/src/raypick/LaserPointer.cpp]: At this point we only have the QVariant rayProps.  RayPickManager is not responsible for parsing the QVariant.  We want to keep all of that work in the scripting interface.

**Pull request merged**